db/migrate/20190514115219_create_firmware_registries.rb

```diff
@@ -0,0 +1,12 @@
+class CreateFirmwareRegistries < ActiveRecord::Migration[5.0]
+  def change
+    create_table :firmware_registries do |t|
+      t.string   :name
+      t.datetime :last_refresh_on
+      t.string   :type
+      t.bigint   :endpoint_id
```

Comment:
Endpoint has a polymorphic `belongs_to :resource` so you don't need the `endpoint_id` here, you just need `has_one :endpoint` in your model